### PR TITLE
Add chance to change password of the account

### LINKED ISSUE #1285: Add chance to change/reset the password of the account
We could add an additional section for the user where it is possible to change/reset their password. The "API access" section could be used for inspiration

### PR DESCRIPTION
**Pull request by shivam-Purohit**
closes #1285 

# Description

add a feature to allow the user to change their password 

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [ ] A new plugin (analyzer, connector, visualizer or playbook) was added or changed, in which case:
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook in `playbook_config.json`.
    - [ ] Check if it could make sense to add that analyzer/connector to other freely available playbooks.
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](./Contribute.md)).
- [x] If the GUI has been modified:
    - [x] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

### CODE REVIEW

**Comment by shivam-Purohit** [comment]
ubuntu project  Running  - Oracle VM VirtualBox 06-06-2023 20_20_56
ubuntu project  Running  - Oracle VM VirtualBox 06-06-2023 20_20_41

**Comment by shivam-Purohit** [comment]
The code was able to change the password successfully. I don't know if the code is good enough. Suggest some changes!

**Review by 0ssigeno — APPROVED** [review]
Backend side lgtm

File: authentication/views.py [hunk 1]
```diff
@@ -101,6 +103,31 @@ def post(self, request, *args, **kwargs):
...
+    @staticmethod
+    def post(request: Request) -> Response:
+        # Get the old password and new password from the request data
+        old_password = request.data.get("old_password")
+        new_password = request.data.get("new_password")
+
+        # Check if the old password matches the user's current password
+        user = request.user
+        if not check_password(old_password, user.password):
+            # Return an error response if the old password doesn't match
```
Comment (mlodic) [inline authentication/views.py:118]: could you add an info log here?

**Review by mlodic — APPROVED** [review]
seems pretty clean.

could you just put the "change password" section before the "plugin configuration" section? I think it makes more sense in terms of usability because this section is a concept very similar to the API sessions one

File: authentication/views.py [hunk 2]
```diff
@@ -115,6 +115,7 @@ def post(request: Request) -> Response:
         # Check if the old password matches the user's current password
         user = request.user
         if not check_password(old_password, user.password):
+            logger.info("Invalid old password")
```
Comment (mlodic) [inline authentication/views.py:118]: could you please log the user too?

Comment (mlodic) [reply authentication/views.py:118]: otherwise it's impossible to debug that behavior and the log is almost useless

Comment (shivam-Purohit) [reply authentication/views.py:118]: logger.info(f"`{uname}` has inputted invalid old password")
will this work
or something like
logger.info(f"ChangePasswordView:`{uname}` inputted invalid old password")

Comment (mlodic) [reply authentication/views.py:118]: first one is enough! ty!

**Comment by mlodic** [comment]
great! :)

**Pull request merged**